storage/local/frankenstein.go

```diff
@@ -273,32 +310,56 @@ func (i *Ingestor) Stop() {
 }
 
 func (i *Ingestor) loop() {
-	defer i.flushAllSeries(true)
+	defer i.flushAllUsers(true)
 	defer close(i.done)
 	tick := time.Tick(flushCheckPeriod)
 	for {
 		select {
 		case <-tick:
-			i.flushAllSeries(false)
+			i.flushAllUsers(false)
 		case <-i.quit:
 			return
 		}
 	}
 }
 
-func (i *Ingestor) flushAllSeries(immediate bool) {
+func (i *Ingestor) flushAllUsers(immediate bool) {
 	if i.chunkStore == nil {
 		return
 	}
-	for pair := range i.fpToSeries.iter() {
-		if err := i.flushSeries(pair.fp, pair.series, immediate); err != nil {
+
+	userIDs := []string{}
```

Comment:
If you know the length beforehand, avoid unnecessary allocs:

```go
userIDs = make([]string, 0, len(i.userState))
```

...but then of course this needs to happen under the lock.